lightning_client.go

```diff
@@ -154,6 +154,11 @@ type LightningClient interface {
 
 	// NetworkInfo returns stats regarding our view of the network.
 	NetworkInfo(ctx context.Context) (*NetworkInfo, error)
+
+	// SubscribeInvoices allows a client to subscribe to updates
+	// of newly added/settled invoices.
+	SubscribeInvoices(ctx context.Context, req InvoiceSubscription) (
+		<-chan *lnrpc.Invoice, <-chan error, error)
```

Comment:
I think we have a lndclient `Invoice` type which we could use here? Where possible, it's nice to convert the rpc structs to lndclient ones so that we can use golang types  - eg `time.Time` rather than unix ts.

Reply:
Just found it, changing my code to use Invoice with unmarshalInvoice.